--- a/agnoster.zsh-theme	2025-04-23 11:05:25.560577124 +0200
+++ b/agnoster.zsh-theme	2025-04-23 11:09:15.194443358 +0200
@@ -36,12 +36,12 @@
 
 case ${SOLARIZED_THEME:-dark} in
     light)
-      CURRENT_FG='white'
-      CURRENT_DEFAULT_FG='white'
+      CURRENT_FG=${CURRENT_FG:-'white'}
+      CURRENT_DEFAULT_FG=${CURRENT_DEFAULT_FG:-'white'}
       ;;
     *)
-      CURRENT_FG='black'
-      CURRENT_DEFAULT_FG='default'
+      CURRENT_FG=${CURRENT_FG:-'black'}
+      CURRENT_DEFAULT_FG=${CURRENT_DEFAULT_FG:-'default'}
       ;;
 esac
 
@@ -308,20 +308,7 @@
     # Git repo and inline path enabled, hence only show the git root
     prompt_segment "$AGNOSTER_DIR_BG" "$AGNOSTER_DIR_FG" "$(git_toplevel | sed "s:^$HOME:~:")"
   else
-    prompt_segment "$AGNOSTER_DIR_BG" "$AGNOSTER_DIR_FG" "$(short_pwd)"
-  fi
-}
-
-short_pwd() {
-  local dir="${PWD/#$HOME/~}"
-  local segments=("${(s:/:)dir}")
-  local count="${#segments[@]}"
-  local max_dirs=3
-  if (( count > max_dirs )); then
-    local shortened=("${segments[@]: -max_dirs}")
-    print -n "…/${(j:/:)shortened}"
-  else
-    print -n "$dir"
+    prompt_segment "$AGNOSTER_DIR_BG" "$AGNOSTER_DIR_FG" '%~'
   fi
 }
 
